Lib/sysconfig.py

```diff
@@ -58,6 +58,25 @@
         },
     }
 
+_get_preferred_schemes = None
+
+
+def _load_vendor_schemes():
+    # add vendor defined schemes
+    try:
+        import _vendor_config
+
+        _INSTALL_SCHEMES.update({
+            name: scheme
+            for name, scheme in _vendor_config.EXTRA_INSTALL_SCHEMES.items()
+            if name not in _INSTALL_SCHEMES
```

Comment:
If this check was not here, we could use this instead of our patch entirely. I'm not sure what is the benefit of the check.

Reply:
That breaks several assumptions in other parts of the code, so I choose to not go into that right now. We could remove it in the future if it there is a valid use case, though I can't really think of anything strong enough to warrant that.

Reply:
FFY00, are you saying that removing the check would break several assumptions in other parts of the code? Can you provide an example of such a broken assumption? Isn't "vendor doesn't have to patch sysconfig" a compelling reason to warrant it (given that's the goal of this patch)?

Reply:
> FFY00, are you saying that removing the check would break several assumptions in other parts of the code? Can you provide an example of such a broken assumption?

There are parts of the `site`, `venv` and `distutils` modules, for example, that hardcode `site-packages`. Allowing vendors to change this path would cause at the minimum a moderate amount of breakage.

> Isn't "vendor doesn't have to patch sysconfig" a compelling reason to warrant it (given that's the goal of this patch)?

It is, though, generally there are several parts of the code that can be change to solve such issues, and I think there might be easier, and probably better, ways to do it rather than allowing vendors to change the default install schemes.

So the question here should be, why would vendors need to change the default schemes? Can we solve the issue in a different way? What would make more sense, removing this check, or any of the alternatives?

Fedora's case can be solved by changing `distutils.command.install` to respect `get_preferred_schemes`, which I feel is a much better solution.
If there are any other reasons where someone might want to remove this check, we can look at them and decide what makes the most sense.